starknet_py/net/account/account_client.py

```diff
@@ -449,3 +472,38 @@ def parse_call(
         calldata.append(data)
 
     return [calldata, entire_calldata]
+
+
+def invoke_function_by_version(
+    # pylint: disable=too-many-arguments
+    account_contract_address: AddressRepresentation,
+    calldata: List[int],
+    signature: List[int],
+    max_fee: int,
+    version: int,
+    nonce: int,
+    entry_point_selector: int,
+) -> InvokeFunction:
+    common_params = {
+        "calldata": calldata,
+        "signature": signature,
+        "max_fee": max_fee,
+        "version": version,
+    }
+
+    if version == 0:
+        return InvokeFunction(
+            contract_address=account_contract_address,
+            entry_point_selector=entry_point_selector,
+            **common_params,
+        )
+
+    # remove when new cairo will be released
+    # pylint: disable=unexpected-keyword-arg
+    return InvokeFunction(
+        account_contract_address=account_contract_address, nonce=nonce, **common_params
+    )
```

Comment:
This is pure speculation, but for cairo-lang 0.10 will probably have to provide different dataclasses for InvokeFunction version 0 and 1 if their fields differ. This will require manual verification once we get access to new cairo-lang.